### ISSUE TITLE
Missing setup.py?

### ISSUE THREAD

**Issue by uahic** [facebookresearch/Mask2Former#119]
Quick question from a detectron noob:

Is there a certain way to 'register' a project within detectron2 or is detectron2 just supposed to be used as a library?
I noticed that (at least) in maskformer 1+2 there is no setup.py and therefor an import via 'import mask2former' is not possible straight away.

Sure, you can import the 'sys' library and append the path of Mask2Former but either you need config files or hardcode  absolute paths into your code which doesnt make them very portable. Another issue is the python auto completion (e.g. visual studio code). This doesnt work when sys.path is extended. The only 2 ways there is to create .env files (containing PYTHONPATH) and tell VSCode to load it or before starting vscode to export PYTHONPATH on the shell. As I am using conda and virtualenv this is also suboptimal.

So then the question: why is there no proper setup.py? is this intentional and have I missed mechanics related to detectron2? or did people simply get along without so far and nobody cared to create one?

**Comment by uahic** [comment]
Appendix:
If a setup.py is not an option, then conda users might consider the conda-build package (https://docs.conda.io/projects/conda-build/en/latest/resources/commands/conda-develop.html) and simply let this tool add a .pth file in the corresponding conda environment, e.g. ``` conda-develop /data/repos/MaskFormer2 ```. Works like charm
